tests/test_nurse_scheduling.py

```diff
@@ -31,11 +31,9 @@ def test_smoke(self):
         energy_found, _ = number_etc.split('\\nChecking', 1)
         str2 = 'Checking Hard shift constraint  0.0'
         str3 = 'Checking Hard nurse constraint  0.0'
-        str4 = 'Checking Soft nurse constraint  0.6'
-        self.assertAlmostEqual(energy, float(energy_found))
+        self.assertTrue(0.5999 < float(energy_found) <= 0.6001)
```

Comment:
Better to use [`assertAlmostEqual`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertAlmostEqual) with `places=4`.

Reply:
While at it, you might want to fix a few more issues: use `assertIn` below, and `sys.executable` instead of hard-coded python.